runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -5298,12 +5298,11 @@ J9::Z::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node * node, TR::CodeGenerator
 
    if (!sourceChild->isNonNull())
       {
-      // If we have an actionable wrtbar and a NULL ptr, branch around the wrt bar
-      // as it needs not be exec'd
-      TR::InstOpCode::Mnemonic compareAndBranchMnemonic = usingCompressedPointers ? TR::InstOpCode::C : TR::InstOpCode::getCmpOpCode();
-
-      generateS390CompareAndBranchInstruction(cg, compareAndBranchMnemonic, node, srcReg, 0, TR::InstOpCode::COND_BE, (doWrtBar || doCrdMrk)?simpleStoreLabel:wbLabel, false, true);
+      // Note the use of 64-bit compare for compressedRefs and use of the decompressed `srcReg` register
+      // Compare object with NULL. If NULL, branch around write barrier as it's not required
```

Comment:
The comment should probably also state it branches around the array store checks, and potential write barrier / card marking logic.

Reply:
@klangman let's make that update to the comment. I'll skip the tests because we already ran them. For record here is the link:

https://ci.eclipse.org/openj9/job/PullRequest-OpenJ9/5869/